packages/compass-import-export/src/modules/import.ts

```diff
@@ -353,63 +392,98 @@ export const startImport = () => {
       });
     }
 
-    promise
-      .finally(() => {
-        progressCallback.flush();
-      })
-      .then((result) => {
-        track('Import Completed', {
-          duration: Date.now() - startTime,
-          file_type: fileType,
-          all_fields: exclude.length === 0,
-          stop_on_error_selected: stopOnErrors,
-          number_of_docs: result.docsWritten,
-          success: true,
-        });
+    let result: ImportResult;
+    try {
+      result = await promise;
+    } catch (err: any) {
+      track('Import Completed', {
+        duration: Date.now() - startTime,
+        file_type: fileType,
+        all_fields: exclude.length === 0,
+        stop_on_error_selected: stopOnErrors,
+        number_of_docs: err.result.docsWritten,
+        success: !err,
+      });
+
+      log.error(mongoLogId(1001000081), 'Import', 'Import failed', {
+        ns,
+        errorLogFilePath,
+        docsWritten: err.result.docsWritten,
+        error: err.message,
+      });
+      debug('Error while importing:', err.stack);
+
+      showFailedToast(err);
+
+      return dispatch(onFailed(err));
+    } finally {
+      importDone = true;
+      errorLogWriteStream?.close();
+      progressCallback.flush();
```

Comment:
I don't think you want flush here for the case when the error happened though, it's convenient to handle both in finally, but stopping import (whether by aborting or error being thrown in the process) should stop reporting progress too, so why would you want to flush these especially after doing all the error handling that affects the state?

Reply:
Makes sense, updated to flush only on success, and updated both `importCSV` and `importJSON` to not report progress when they've been aborted.
A bit unrelated, also removed some unused state related to the progress which wasn't being used anymore.